app/controllers/application_controller.rb

```diff
@@ -2,9 +2,115 @@
 
 class ApplicationController < Sinatra::Base
 
-  configure do
-    set :public_folder, 'public'
-    set :views, 'app/views'
-  end
+ 	configure do
+	    set :public_folder, 'public'
+	    set :views, 'app/views'
+	    enable :sessions
+		set :session_secret, "password_security"
+	end
+
+	get "/" do
+		erb :index
+	end
+
+	get "/signup" do
+		erb :signup
+	end
+
+	post "/signup" do
+	    if !params[:username].empty? && !params[:email].empty? && !params[:password].empty?
+	    	user = User.create(:username => params[:username], :email => params[:email], :password => params[:password])
+	    	session[:user_id] = user.id
+	    	redirect "/tweets"
+		else
+	    	redirect "/signup"
+		end
+	end
+
+
+	get "/login" do
+		erb :'users/login'
+	end
+
+	post "/login" do
+		user = User.find_by(:username => params[:username])
+	    if user && user.authenticate(params[:password])
+	        session[:user_id] = user.id
+	        redirect "/tweets"
+	    else
+	        redirect "/users/login"
+	    end
+	end
+
+	get "/logout" do
+		session.clear
+		redirect "/login"
+	end
+
+	get "/tweets" do
+		@tweets = Tweet.all
+		erb :'tweets/tweets'
+	end
+
+	get "/tweets/new" do
+		erb :'tweets/create_tweet'
+	end
+
+	post "/tweets/new" do
```

Comment:
Typically, tweet creation would be posted to `/tweets` instead of `/tweets/new`.